hail/src/main/scala/is/hail/expr/ir/ForwardLets.scala

```diff
@@ -68,17 +68,19 @@ object ForwardLets {
       ir match {
         case Let(name, value, body) =>
           val refs = uses.lookup(ir)
-          if (shouldForward(value, refs, ir))
-            rewrite(body, env.bindEval(name -> rewrite(value, env)))
+          val rewriteValue = rewrite(value, env)
+          if (shouldForward(rewriteValue, refs, ir))
+            rewrite(body, env.bindEval(name -> value))
           else
             mapRewrite()
```

Comment:
This double-rewrites value.  I think `Let(name, rewrite(body, env), rewriteValue)` better?

Reply:
ah, yeah.

Reply:
silly mistake.

Reply:
fixed now, though